src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -195,11 +202,103 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
             }
 
             Response.StatusCode = 401;
-            Response.Headers.Append(HeaderNames.WWWAuthenticate, Options.Challenge);
+
+            var errorDescription = CreateErrorDescription(eventContext.AuthenticateFailure);
```

Comment:
Have you considered adding new `Error`/`ErrorDescription` properties to the challenge context and computing the error description before invoking the `Challenge` event? This way, one could easily replace the error code/description without having to manually create the `WWW-Authenticate` header.